# HG changeset patch
# User notnoop
# Date 1267155668 18000
# Node ID 48f8ae25e672f2afe4f343a9b9d09114959ddde4
# Parent  e18a31852290b8a8992a1e68cdba574489ffc9c1
add an ant build file to automate build and parser generation

diff --git a/build.xml b/build.xml
new file mode 100644
--- /dev/null
+++ b/build.xml
@@ -0,0 +1,44 @@
+<!--
+  This is an Ant build file for compiling the Java Parser
+-->
+<project name="javaparser" default="dist" basedir=".">
+
+    <description>
+      Builds the java parser for JSR 308 index files.
+    </description>
+
+    <property name="src" value="src"/>
+    <property name="build" value="build"/>
+
+    <property file="build.properties"/>
+    <property environment="env"/>
+
+    <property name="javacc.home" value="${env.JAVACC_HOME}"/>
+    <property name="dist.file" value="javaparser.jar"/>
+
+    <target name="prep" description="Create required directories">
+        <mkdir dir="${build}"/>
+    </target>
+
+    <target name="clean" description="Remove generated files">
+        <delete dir="${build}"/>
+        <delete file="${dist.file}"/>
+    </target>
+
+    <target name="gen-parser" description="Generate parser files">
+        <javacc target="src/japa/parser/java_1_5.jj"
+                javacchome="${javacc.home}" />
+    </target>
+
+    <target name="build" depends="prep,gen-parser" description="Compile files">
+        <javac destdir="${build}">
+          <src path="${src}"/>
+          <include name="**/*.java"/>
+        </javac>
+    </target>
+
+    <target name="dist" depends="build" description="Create jar file">
+        <jar destfile="${dist.file}" basedir="${build}"/>
+    </target>
+
+</project>
